src/game/main.cpp

```diff
@@ -162,6 +173,201 @@ void Check_Windowed(int argc, char *argv[])
+    }
+
+    CopyProtect::checkForMessage(message, l_param);
+
+    switch (message) {
+        case WM_SETCURSOR:
+            if (g_theWin32Mouse != nullptr) {
+                g_theWin32Mouse->Set_Cursor(g_theWin32Mouse->Get_Mouse_Cursor());
+            }
+            return TRUE;
+
+        case WM_SIZE:
+            if (!g_creatingWindow) {
+                g_gameNotFullscreen = false;
+            }
+            break;
+
+        case WM_ACTIVATE:
+            switch (LOWORD(w_param)) {
+                case WA_INACTIVE:
+                    ClipCursor(nullptr);
+                    if (g_theAudio != nullptr) {
+                        g_theAudio->Lose_Focus();
+                    }
+                    break;
+                default:
+                case WA_ACTIVE:
+                case WA_CLICKACTIVE:
+                    if (g_theMouse != nullptr) {
+                        g_theMouse->Set_Mouse_Limits();
+                    }
+                    if (g_theAudio != nullptr) {
+                        g_theAudio->Regain_Focus();
+                    }
+                    break;
+            }
+            break;
+
+        case WM_SETFOCUS:
+            if (g_theKeyboard != nullptr) {
+                g_theKeyboard->Reset_Keys();
+            }
+            if (g_theWin32Mouse != nullptr) {
+                g_theWin32Mouse->Set_Ignore_Events(false);
+            }
+            break;
+
+        case WM_KILLFOCUS:
+            if (g_theKeyboard != nullptr) {
+                g_theKeyboard->Reset_Keys();
+            }
+            if (g_theWin32Mouse != nullptr) {
+                g_theWin32Mouse->Set_Ignore_Events(true);
+            }
+            break;
+
+        case WM_PAINT:
+            if (g_gameNotFullscreen) {
+                PAINTSTRUCT ps{};
+                auto hDC = BeginPaint(window_handle, &ps);
+                if (g_splashImage != nullptr) {
+                    const auto old_dc = SaveDC(hDC);
+                    auto compat_dc = CreateCompatibleDC(hDC);
+                    auto ha = SelectObject(compat_dc, g_splashImage);
+                    BitBlt(hDC, 0, 0, 800, 600, compat_dc, 0, 0, SRCCOPY);
+                    SelectObject(compat_dc, ha);
+                    DeleteDC(compat_dc);
+                    RestoreDC(hDC, old_dc);
+                }
+                EndPaint(window_handle, &ps);
+                return 0; // BUGFIX: vanilla returned 1 WinAPI says return 0
+            }
+            break;
+
+        case WM_CLOSE:
+            if (g_theGameEngine->Get_Quitting()) {
+                if (g_theKeyboard != nullptr) {
+                    g_theKeyboard->Reset_Keys();
+                }
+                if (g_theWin32Mouse != nullptr) {
+                    g_theWin32Mouse->Set_Ignore_Events(false);
+                }
+                break;
+            }
+            g_theMessageStream->Append_Message(GameMessage::MessageType::MSG_META_DEMO_INSTANT_QUIT);
+            return 0;
```

Comment:
It can return `FALSE` for 0 since it already returns `TRUE` for 1.

Reply:
Winapi states that this returns 0 for handling with message.